lib/msf/ui/console/command_dispatcher/creds.rb

```diff
@@ -502,8 +509,8 @@ def creds_search(*args)
     end
 
     if mode == :delete
-      result = framework.db.delete_credentials(ids: matched_cred_ids.uniq)
-      delete_count = result.size
+      delete_count = matched_cred_ids.size
+      result = framework.db.delete_credentials(ids: matched_cred_ids.concat(cracked_cred_ids).uniq)
```

Comment:
Fixed with https://github.com/rapid7/metasploit-framework/pull/17689/commits/28a2bcf9d79c5e406dcf5aa5e5bbd8e3e188febf

Reply:
So Is it fine to show them 6 rows of creds when `creds` command is used and show them 12 creds got deleted while` creds -d`  doesn't it impact the user experience? doesn't the user gets confused that there are only 6 rows but we are 12 rows got deleted?

Reply:
@manishkumarr1017 I'm confused what your thinking happens here. I was under the impression that `framework.db.delete_credentials` may not necessarily delete all the credentials that you pass to it. Therefore you should get the result of this, aka `result.size` and report that as the number of successfully deleted entries.

Reply:
Ok. I understood your concern. but we are showing the number of deleted creds with the end users right? So let's take one sample output.

```
Credentials
===========

host  origin  service  public             private                                                                                              realm  private_type        JtR Format        cracked_password
----  ------  -------  ------             -------                                                                                              -----  ------------        ----------        ----------------
                       oracle12c_epsilon  H:DC9894A01797D91D92ECA1DA66242209;T:E3243B98974159CC24FD2C9A8B30BA62E0E83B6CA2FC7C5517 (TRUNCATED)         Nonreplayable hash  pbkdf2,oracle12c  epsilon
                       SYSTEM             9EEDFA0AD26C6D52                                                                                            Nonreplayable hash  des,oracle        THALES
                       example            md5be86a79bf2043622d58d5453c47d4860                                                                         Postgres md5        raw-md5,postgres  password
                       simon              4F8BC1809CB2AF77                                                                                            Nonreplayable hash  des,oracle        A
                       DEMO               S:8F2D65FB5547B71C8DA3760F10960428CD307B1C6271691FC55C1F56554A;H:DC9894A01797D91D92ECA1 (TRUNCATED)         Nonreplayable hash  raw-sha1,oracle   epsilon
                       oracle11_epsilon   S:8F2D65FB5547B71C8DA3760F10960428CD307B1C6271691FC55C1F56554A;H:DC9894A01797D91D92ECA1 (TRUNCATED)         Nonreplayable hash  raw-sha1,oracle   epsilon

[*] Deleted 12 creds
```

So as you can see there are only 6 rows but it shows the 12 creds are deleted. Does it not confuse the end users? I understood that this might be helpful for us to debug but I feel it certainly confuses the end users.